projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,234 @@
+    private SimpleDateFormat dateAndTimeFormatter;
+    private SimpleDateFormat dateFormatter;
+
+    // Initialise until App has client-side form
+    private boolean sentHaser = true;
+    private boolean sentMoreThanFiveLawsuits = false; // The past year
+    private Date receivedSpamDate = new Date();
+
+// ------------------------------- Lawsuit Form Fields ------------------------------- //
+    // General
+    private String spamType = "הודעה אלקטרונית";
+    private String claimAmount = "סכום תביעה";
+    private String moreThanFiveLawsuits = "הגיש";
+    private String lessThanFiveLawsuits = "לא הגיש/ו";
+    private String claimCaseHaser = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+    private String claimCaseSubscription = "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש לקבלת דבר/י הפרסומת";
+
+    // User
+    private String userPrivateName = "שם פרטי";
+    private String userLastName = "שם משפחה";
+    private String userID = "ת ז משתמש";
+    private String userAddress = "כתובת משתמש";
+    private String userPhone = "טלפון משתמש";
+    private String userFax = "פקס משתמש";
+
+    // First spam company's details
+    private String companyName = "שם החברה";
+    private String companyId = "מספר ח.פ";
+    private String companyAddress = "כתובת החברה";
+    private String companyPhone = "טלפון החברה";
+    private String companyFax = "פקס חברה";
+
+    // Second spam company's details
+    private String company2Name = "שם החברה 2";
+    private String company2Id = "מספר ח.פ 2";
+    private String company2Address = "כתובת החברה 2";
+    private String company2Phone = "טלפון החברה 2";
+    private String company2Fax = "פקס חברה 2";
+// ---------------------------------------------------------------------------------------:
+    Button createPdfButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_lawsuit_pdf);
+        createPdfButton = (Button) findViewById(R.id.button_createPDF);
+
+        getWriteStoragePermission();
+        initDirStructure();
+
+        dateAndTimeFormatter = new SimpleDateFormat(DATE_TIME_FORMAT);
+        dateAndTimeFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+        dateFormatter = new SimpleDateFormat(DATE_FORMAT);
+        dateFormatter.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+
+        // Generate lawsuit onClick
+        createPdfButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                createPdf();
+            }
+        });
+
+    }
+
+    private void createPdf() {
+
+        // Activate iText license
+        // TODO: Find a better solution for reading the licensekey file (This was a just a "quick fix" to get it working).
+        InputStream licenseIs = getResources().openRawResource(R.raw.itextkey);
+        LicenseKey.loadLicenseFile(licenseIs);
```

Comment:
For the most part, we try not to create local variables if not needed.
Exceptions are when the expression gets too complicated.

So here, I would use:
`LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));`

Reply:
Done.